### PR TITLE
refactor(core): count new users by created_at

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

- [Add 'reated_at' column to 'users' table](https://github.com/logto-io/logto/commit/e28f3cd5aea784fc15a369076dfb67725bff1f87)
- [Analyze new user counts with 'users' table 'created_at' column](https://github.com/logto-io/logto/commit/ef75fbdd2a92c0259347a0757848593bfd56710b) 

Reference: original Slack discussion [thread](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1658922858838919)

## Testing

Locally tested.

### CODE REVIEW

File: packages/core/src/queries/user.ts [hunk 1]
```diff
@@ -158,3 +158,15 @@ export const hasActiveUsers = async () =>
...
+  startTimeExclusive: number,
+  endTimeInclusive: number
+) =>
+  envSet.pool.any<{ date: string; count: number }>(sql`
+    select date(${fields.createdAt}), count(*)
+    from ${table}
+    where ${fields.createdAt} > to_timestamp(${startTimeExclusive}::double precision / 1000)
+    and ${fields.createdAt} <= to_timestamp(${endTimeInclusive}::double precision / 1000)
+    group by date(${fields.createdAt})
+  `);
```
Comment (darcyYe) [inline packages/core/src/queries/user.ts:172]: Should this method be placed at `packages/core/src/lib/dashboard.ts` since it is in the 'dashboard' scope?

Comment (darcyYe) [reply packages/core/src/queries/user.ts:172]: Do we need to filter those users whose `create_at` is null?

Comment (IceHe) [reply packages/core/src/queries/user.ts:172]: created_at is `not null`.

Comment (IceHe) [reply packages/core/src/queries/user.ts:172]: It's better to put SQL queries in `queries/*.ts` grouping by the table.

**Review by wangsijie — APPROVED** [review]
LGTM
nit: try the title: count users by createdAt

File: packages/schemas/alterations/next-1664356000-add-created-at-column-to-users.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { sql } from 'slonik';
+
+import { AlterationScript } from '../lib/types/alteration';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      alter table users add column created_at timestamptz not null default (now());
```
Comment (simeng-li) [inline packages/schemas/alterations/next-1664356000-add-created-at-column-to-users.ts:8]: whenever run the db migrations, all the existing users will become new users? Shouldn't we load old users data from the log?

Comment (IceHe) [reply packages/schemas/alterations/next-1664356000-add-created-at-column-to-users.ts:8]: #2031 Done

**Pull request merged**